### PR TITLE
[tests-only][full-ci]Add tests for list grant role related to `share by me` and `share with me`

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds tests for list grant role related to share with me and share by me feature

**Note** Only limited test cases has been added for additional role 

## Related Issue
- https://github.com/owncloud/ocis/issues/10951

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+  So that users can view activities and grant list of shared resources
+
+  Background:
+    Given these users have been created with default attributes:
+      | username |
+      | Alice    |
+      | Brian    |
+
+
+  Scenario Outline: sharer lists share send by her (Personal space)
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature:14]:
```suggestion
  Scenario Outline: sharer lists share shared by her (Personal space)
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature [hunk 2]
```diff
@@ -0,0 +1,819 @@
...
+        }
+      }
+      """
+    Examples:
+      | permissions-role       |
+      | Viewer With ListGrants |
+      | Editor With ListGrants |
+
+
+  Scenario Outline: sharer list share send by her when same file is shared with multiple user(Personal space)
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature:84]:
```suggestion
  Scenario Outline: sharer list share shared by her when same file is shared with multiple user(Personal space)
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature [hunk 3]
```diff
@@ -0,0 +1,819 @@
...
+                          }
+                        }
+                      }
+                    },
+                    "eTag": {"pattern": "%etag_pattern%"},
+                    "file": {
+                      "type": "object",
+                      "required": ["mimeType"],
+                      "properties": {
+                        "mimeType": {"pattern": "text/plain"}
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature:322]:
```suggestion
                        "mimeType": {"const": "text/plain"}
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature [hunk 4]
```diff
@@ -0,0 +1,819 @@
...
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should contain resource "textfile.txt" with the following data:
+      """
+      {
+        "type": "object",
+        "required": [
+          "parentReference",
+          "permissions",
+          "name",
+          "size"
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantShareRoles.feature:628]: You can minimize.

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature [hunk 5]
```diff
@@ -243,3 +243,806 @@ Feature: ListGrants role
       | permissions-role       |
       | Viewer With ListGrants |
       | Editor With ListGrants |
+
+
+  Scenario Outline: sharer lists shared by her (Personal space)
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature:248]:
```suggestion
  Scenario Outline: sharer lists shared-by-me (Personal space)
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature [hunk 6]
```diff
@@ -243,3 +243,806 @@ Feature: ListGrants role
...
+      | Editor With ListGrants |
+
+
+  Scenario Outline: sharer list share shared by her when same file is shared with multiple user(Personal space)
+    Given the administrator has enabled the permissions role "<permissions-role>"
+    And user "Carol" has been created with default attributes
+    And user "Alice" has uploaded file with content "to share" to "textfile.txt"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | textfile.txt      |
+      | space           | Personal           |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature:324]:
```suggestion
      | resource        | textfile.txt       |
      | space           | Personal           |
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature [hunk 7]
```diff
@@ -243,3 +243,806 @@ Feature: ListGrants role
...
+        }
+      }
+      """
+    Examples:
+      | permissions-role            |
+      | Viewer With ListGrants      |
+      | File Editor With ListGrants |
+
+
+  Scenario Outline: sharee list shared with him (Personal space)
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature:470]:
```suggestion
  Scenario Outline: sharee list shared-with-me (Personal space)
```

File: tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature [hunk 8]
```diff
@@ -243,3 +243,806 @@ Feature: ListGrants role
...
+        }
+      }
+      """
+    Examples:
+      | permissions-role            |
+      | Viewer With ListGrants      |
+      | File Editor With ListGrants |
+
+
+  Scenario: share exists even though sharee has been disabled (Project space)
```
[the remark this answers is not in the document]
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature:914]: thats why i added only 2 edge case and alternate scenario for file and folders

[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNgAdditionalShareRole/listGrantsShareRole.feature:914]: let's not do extreme edge edges for optional roles.

**Review by saw-jan — APPROVED** [review]
